autoload/fzf/vim.vim

```diff
@@ -1243,12 +1243,13 @@ function! fzf#vim#complete(...)
   let s:opts = s:eval(s:opts, 'options', s:query)
   let s:opts = s:eval(s:opts, 'extra_options', s:query)
   if has_key(s:opts, 'extra_options')
-    let s:opts.options =
-      \ join(filter([get(s:opts, 'options', ''), remove(s:opts, 'extra_options')], '!empty(v:val)'))
+    call s:merge_opts(s:opts, remove(s:opts, 'extra_options'))
   endif
   if has_key(s:opts, 'options')
     " FIXME: fzf currently doesn't have --no-expect option
-    let s:opts.options = substitute(s:opts.options, '--expect=[^ ]*', '', 'g')
+    if type(s:opts.options) == s:TYPE.string
+      let s:opts.options = substitute(s:opts.options, '--expect=[^ ]*', '', 'g')
+    endif
```

Comment:
fzf now supports `--no-expect` so we can simplify this part.

Reply:
Can you promote the fix on master so I have a reference on what you have in mind?
I'll rebase on `master` after.

Reply:
The line here is to make sure that there is no `--expect` option specified in the chain of options. Since fzf now has `--no-expect`, we can simply append `--no-expect` at the end of the string (of list).

```diff
diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index c1ca760..f40baef 100644
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -1256,8 +1256,7 @@ function! fzf#vim#complete(...)
       \ join(filter([get(s:opts, 'options', ''), remove(s:opts, 'extra_options')], '!empty(v:val)'))
   endif
   if has_key(s:opts, 'options')
-    " FIXME: fzf currently doesn't have --no-expect option
-    let s:opts.options = substitute(s:opts.options, '--expect=[^ ]*', '', 'g')
+    let s:opts.options = .' --no-expect'
   endif
 
   call feedkeys("\<Plug>(-fzf-complete-trigger)")
```